ui/app/src/App.tsx

```diff
@@ -122,6 +150,25 @@ function App() {
           )}
         </>
       </div>
+      {stream && (
+        <div>
+          <CancelStream
+            streamId={stream.streamId}
+            reason={reason}
+            creator={stream.creator}
+            sm={sm}
+          />
+        </div>
+      )}
+      {stream && (
```

Comment:
suggestion(non-blocking): I wonder if this should be more intelligent.  Checking against `max_stream_life` to make sure an update would succeed.  This check is what I'm concerned about:

https://github.com/ApeWorX/ApePay/blob/61ad4b17996723b5a806aa21400a33b32c8d28af/contracts/StreamManager.vy#L201-L204

Maybe something to deal with later.

Reply:
Spoke with @fubuloubu ; I will create an issue; max stream life isn't implemented yet so I'll remove it from this PR